src/vsr/client.zig

```diff
@@ -93,6 +99,10 @@ pub fn ClientType(comptime StateMachine_: type, comptime MessageBus: type) type
         /// Used for end-to-end keepalive, and to discover a new primary between requests.
         ping_timeout: vsr.Timeout,
 
+        /// The round-trip time (estimated by the latest ping/pong pair) from each replica.
+        replica_round_trip_times_ns: [constants.replicas_max]?u64 =
+            .{null} ** constants.replicas_max,
```

Comment:
Hm, so ideally we should clear this periodically. Imagine the case where we had a replica real close, and then that replica dies, and all our other pings are slower. The false fast ping would lead us to an incorrect estimation, until we hear from the replica agiain. 

I guess this is mostly mitigated by us taking the median, but what bothers me is just the fact that we use different logic between the replica and the client.

🤔 can we use the full `Clock` here? I _think_ client can be treated as a stand-by? They need the same logic, where they are aware of cluster time, but don't affect it.  That way, we'll even dedup the median-computing code.

Reply:
:thinking: `Clock` additionally demands the replica's wall time, which we don't include on pongs.

Reply:
Ah, right, then I think it is fine as is!